### PR TITLE
Code Re-Use

### LINKED ISSUE #16: Rethinking the `_targets.R` approach
The currently used approach to displaying and then running targets workflows in the lesson is to:
* Include the `_targets.R` file in the Rmd file, but to set it to `eval=FALSE`
* Add a copy of that script to the character vector in `write_example_plan`
* Add another chunk after the first, with `echo=FALSE`, which uses `write_example_plan()` to write and execute a copy of the workflow. 

An example can be seen here: https://github.com/carpentries-incubator/targets-workshop/blob/4588d719f590d134aea783152cd1bd4c7695f012/episodes/branch.Rmd#L69-L101.

It occurred to me that we could maybe improve this by using some knitr features, and avoid the issue of needing to synchronise the two copies of the script.

First we move all the example workflows (`_targets.R`) files into a directory, let's say `episodes/files/plan_3.R`. Then in the Rmd we use the `file=` option to knitr, which loads the chunk from a file on disk and displays it. This does work with sandpaper, I've verified it myself:
``````
```{r, file="episodes/files/plan_3.R"}
```
``````

Then, when we need to run the workflow, we add an invisible chunk (similar to before), but this time it looks like:
```R
 tar_dir({ 
   tar_make(script="episodes/files/plan_3.R")
 }) 
```

This simplifies both of the chunks quite a lot, should increase maintainability, and makes it easier to edit the sample workflows.
Thoughts?

### PR DESCRIPTION
**Pull request by multimeric**
* Alternative to #17
* The theme of this PR is refactoring the episodes to re-use code, and you will notice the net reduction in about ~500 LoC
* Write workflows from chunks directly into `_targets.R` wherever possible, which simplifies the codebase, for example: https://github.com/carpentries-incubator/targets-workshop/blob/7aa418cdc0da6ac6e17482acbd8f70a0f3bfe48a/episodes/branch.Rmd#L166-L167
* Workflows that are re-used between files have moved into `files/plans`, with one plan per file.
* User-facing utility functions like `clean_penguin_data` have moved into `files/tar_functions`
* Then, whenever we want to show a script to a user (which might be a plan, or a utility function), we just add a `file:` chunk such as: https://github.com/carpentries-incubator/targets-workshop/blob/7aa418cdc0da6ac6e17482acbd8f70a0f3bfe48a/episodes/branch.Rmd#L317
* Whenever we want to show a *subset* of a script to a user, we use `readLines(file)[lines]`: https://github.com/carpentries-incubator/targets-workshop/blob/7aa418cdc0da6ac6e17482acbd8f70a0f3bfe48a/episodes/branch.Rmd#L261
* Add `pushd` and `popd` utility functions that let us reduce the number of hidden chunks significantly, because we can instead just hide the utility functions: https://github.com/carpentries-incubator/targets-workshop/blob/7aa418cdc0da6ac6e17482acbd8f70a0f3bfe48a/episodes/branch.Rmd#L343-L351
* Thus, there is only one copy of each script used throughout the repo, which is very easy to edit.
* Closes #16.

### CODE REVIEW

**Comment by joelnitta** [comment]
@multimeric Thanks, I really like how this cleans up the code. I took it one step further and used this approach for almost all the penguin examples - I think it makes more sense once you get in the mental state of looking at the plan_*.R files.

(BTW the rubber ducky "carpentries-bot" is me... for some odd reason sandpaper repos have a habit of doing that!)

If this looks good to you, I'm happy to merge.

**Comment by multimeric** [comment]
Great! The shame is that if you preferred the files method to the chunk writing method, #17 may have been better to work off. Ah well, at least you've been able to see both approaches this way.

File: episodes/parallel.Rmd [hunk 1]
```diff
@@ -151,62 +112,27 @@ Add these functions to `functions.R` (you can copy-paste the original ones, then
...
+#| echo: false
+plan_10_dir <- make_tempdir()
+pushd(plan_10_dir)
+# Non-slow functions
+write_example_plan("plan_9.R")
+tar_make(reporter = "silent")
+# Slow functions
+write_example_plan("plan_10.R")
+tar_make()
+popd()
```
Comment (multimeric) [inline episodes/parallel.Rmd:135]: fyi you don't need to use `pushd` and `popd` when the whole chunk is hidden, `tar_dir()` etc would be fine. I added them only for the scenario where we want to show *part* of the chunk and hide the directory manipulation, using `echo: [X, Y]`. However if you want to use `pushd` and `popd` universally for consistency, that's fine too.

File: episodes/files/plans/plan_10.R [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    glance_with_mod_name_slow(models),
+    pattern = map(models)
+  ),
+  # Get model predictions
+  tar_target(
+    model_predictions,
+    augment_with_mod_name_slow(models),
+    pattern = map(models)
+  )
+)
```
Comment (multimeric) [inline episodes/files/plans/plan_10.R:42]: A couple of missing terminating newlines in the plans. Setting `"files.insertFinalNewline": true` in VS Code was lifechanging for me

File: episodes/parallel.Rmd (hunk 1 above)
Comment (joelnitta) [reply episodes/parallel.Rmd:135]: Yes, this was for consistency. Leaving as-is.

File: episodes/files/plans/plan_10.R (hunk 2 above)
Comment (joelnitta) [reply episodes/files/plans/plan_10.R]: Thanks for the tip! Fixed, I hope.

**Comment by multimeric** [comment]
If you don't like the chunk saving approach, I can remove it from the codebase to simplify things?

**Comment by joelnitta** [comment]
I still use it in some places, e.g. `write_example_plan(chunk = "example-file-show-1")` in `files.Rmd` (if that's what you're talking about), so I think we should leave it.

**Comment by multimeric** [comment]
Okay, I'm fine with whichever combination of approaches you're happy to work with. Either way removes the redundancy and makes development easier so I'm happy :+1:

**Comment by joelnitta** [comment]
I realized the external plan file works well for the penguins examples, because they get run repeatedly and gradually build up in complexity. But the chunk approach is nice for short "one-off" examples where it's easier to grok with the plan is right there in the Rmd. So thanks for making both available!

**Comment by multimeric** [comment]
Cool, I'm happy to merge then

**Pull request merged**